src/niscope/metadata/functions_addon.py

```diff
@@ -4,4 +4,146 @@
 functions_override_metadata = {
 }
 
+functions_additional_get_self_cal_last_date_and_time = {
+    'FancyGetLastSelfCalDateAndTime': {
+        'codegen_method': 'python-only',
+        'documentation': {
+            'description': 'Returns the date and time of the last self calibration performed.',
+        },
+        'method_templates': [
+            {
+                'documentation_filename': 'default_method',
+                'method_python_name_suffix': '',
+                'session_filename': 'fancy_cal_last_date'
+            }
+        ],
+        'parameters': [
+            {
+                'direction': 'in',
+                'documentation': {
+                    'description': 'Identifies a particular instrument session. You obtain the **vi** parameter from niScope_init.'
+                },
+                'name': 'vi',
+                'type': 'ViSession'
+            },
+            {
+                'direction': 'out',
+                'documentation': {
+                    'description': 'Indicates the **datetime** of the last calibration.'
+                },
+                'name': 'datetime',
```

Comment:
Doesn't really matter.. but `datetime` is the name of a Python library, better not use that for a variable name. Maybe `last_cal_datetime`?